### PR TITLE
Fixes codearena #95

### PR DESCRIPTION
**Pull request by naps62**
Fixes https://github.com/code-423n4/2022-01-sandclock-findings/issues/95

### CODE REVIEW

File: contracts/Vault.sol [hunk 1]
```diff
@@ -169,7 +169,7 @@ contract Vault is
 
     /// See {IVault}
     function yieldFor(address _to)
-        external
+        public
```
Comment (gabrielpoca) [inline contracts/Vault.sol:172]: Is this change intentional?

Comment (naps62) [reply contracts/Vault.sol:172]: I think it came from reverting the `BaseVault` commit
it was previously external, and delegated to `_yieldFor`, which doesn't exist here
